Position subviews correctly when cell align is set.

DawnWright opened the pull request:
Scenario:

Subview in a horizontal layout is placed inside a view that has a fixedDesiredHeight smaller than its desired height. When valign is set to top on the parent, the subview is correctly vertically aligned within the smaller fixedDesiredHeight. But when cellVAlign is set to top on the subview, the subview is vertically placed within the larger desired height. This is because positionSubview is using the wrong cellBounds.

Example:

```
    WeView *yellowView = [[WeView alloc] init];
    yellowView.backgroundColor = [UIColor yellowColor];
    yellowView.fixedDesiredHeight = 300;
    [view addSubviewWithCustomLayout:yellowView];
    
    UIImageView *redImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithColor:[UIColor redColor] size:CGSizeMake(50, 400)]];
    UIImageView *blueImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithColor:[UIColor blueColor] size:CGSizeMake(50, 100)]];
    
    [[yellowView addSubviewsWithHorizontalLayout:@[redImageView, blueImageView]]
     setVAlign:V_ALIGN_TOP];
```

Yields correct top-aligned layout:
![screenshot 2015-03-06 12 18 56](https://cloud.githubusercontent.com/assets/5578654/6535758/4a79a00e-c3fc-11e4-845b-b9b9cc2116c0.png)

```
    WeView *yellowView = [[WeView alloc] init];
    yellowView.backgroundColor = [UIColor yellowColor];
    yellowView.fixedDesiredHeight = 300;
    [view addSubviewWithCustomLayout:yellowView];
    
    UIImageView *redImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithColor:[UIColor redColor] size:CGSizeMake(50, 400)]];
    UIImageView *blueImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithColor:[UIColor blueColor] size:CGSizeMake(50, 100)]];
    
    [[yellowView addSubviewsWithHorizontalLayout:@[redImageView, blueImageView]]
     setVAlign:V_ALIGN_CENTER];
```
Yields correct center-aligned layout:
![screenshot 2015-03-06 12 20 47](https://cloud.githubusercontent.com/assets/5578654/6535750/421edfdc-c3fc-11e4-8fe4-7c8576709788.png)

```
    WeView *yellowView = [[WeView alloc] init];
    yellowView.backgroundColor = [UIColor yellowColor];
    yellowView.fixedDesiredHeight = 300;
    [view addSubviewWithCustomLayout:yellowView];
    
    UIImageView *redImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithColor:[UIColor redColor] size:CGSizeMake(50, 400)]];
    redImageView.cellVAlign = V_ALIGN_TOP;
    UIImageView *blueImageView = [[UIImageView alloc] initWithImage:[UIImage imageWithColor:[UIColor blueColor] size:CGSizeMake(50, 100)]];
    
    [[yellowView addSubviewsWithHorizontalLayout:@[redImageView, blueImageView]]
     setVAlign:V_ALIGN_CENTER];
```
Yields incorrect layout of red block:
![screenshot 2015-03-06 12 20 47](https://cloud.githubusercontent.com/assets/5578654/6535782/7a7d87ca-c3fc-11e4-9e21-6c9b235e80c5.png)

With fix applied, the new result properly aligns red to top:
![screenshot 2015-03-06 12 31 38](https://cloud.githubusercontent.com/assets/5578654/6535830/d2df6898-c3fc-11e4-8b98-ae28061cee25.png)

PTAL @charlesmchen

charlesmchen commented:
Hmm, I'm not sure I understand.  Can we discuss when you get in?

DawnWright commented:
Updated the fix and provided an example of the bug. @charlesmchen

charlesmchen commented:
LGTM

charlesmchen commented:
@DawnWright thanks for the great PR documentation.  When you merge please update both forks.

DawnWright commented:
The fix was correct for the cross-axis direction, but not for axis direction. I've updated the change here: https://github.com/FiftyThree/WeView2/commit/c9a934734c571a66c831b277d851291a1c595c8e

I also cleaned up an unused param on existing API.

PTAL @charlesmchen

charlesmchen commented on WeView/Layouts/WeViewLayout.h:
```diff
@@ -79,6 +79,12 @@
+
+// The vertical alignment of a subview in this layout.
+- (VAlign)subviewCellVAlign:(UIView *)subview;
+
```
Seems like these belong in `WeViewLayout+Subclass.h`?

DawnWright replied:
Ah, I didn't know about that file - fixed!

charlesmchen commented on WeView/Layouts/WeViewLinearLayout.m:
```diff
@@ -367,43 +367,39 @@ - (void)dumpItemSizes:(NSString *)label
-- (void)hAlignIndex:(CGFloat *)index
-         extraSpace:(CGFloat)extraSpace
-               view:(UIView *)view
+- (CGFloat)hAlignIndex:(CGFloat)index
```
I wish I hadn't used the term index - it's very confusing.

charlesmchen commented:
LGTM

This turned into a big change.  I wish I had time to test.

DawnWright commented:
> I wish I had time to test.

No worries, I'm testing things (hence the followup commit), and I'm going to run this locally for a bit before committing.

DawnWright commented:
Reopening this against FiftyThree fork: https://github.com/FiftyThree/WeView2/pull/1
